beacon-chain/sync/subscriber.go

```diff
@@ -247,6 +250,13 @@ func (r *Service) subscribeDynamicWithSubnets(
 				}
 				for _, idx := range wantedSubs {
 					if _, exists := subscriptions[idx]; !exists {
+						// do not subscribe if we have no peers in the same
+						// subnet
+						topic := p2p.GossipTypeMapping[reflect.TypeOf(&pb.Attestation{})]
+						numOfPeers := r.p2p.PubSub().ListPeers(fmt.Sprintf(topic, idx))
+						if len(r.p2p.Peers().SubscribedToSubnet(idx)) == 0 && len(numOfPeers) == 0 {
```

Comment:
Perhaps we can log a debug warning here if this occurs, it will help us debug whether or not the feature is working in prod